lib/main/STM32F7xx_HAL_Driver/Inc/stm32f7xx_ll_spi.h

```diff
@@ -1375,6 +1375,7 @@ __STATIC_INLINE uint16_t LL_SPI_ReceiveData16(SPI_TypeDef *SPIx)
   */
 __STATIC_INLINE void LL_SPI_TransmitData8(SPI_TypeDef *SPIx, uint8_t TxData)
 {
+  /* TODO: remove the warning */
   *((__IO uint8_t *)&SPIx->DR) = TxData;
 }
```

Comment:
It's treating a uint32_t as a pointer to uint8_t - but dereferencing so at first glance I think we can just use SPIx pointer plus distance to DR as the uint8 location - tricking the compiler into ignoring it.